pyaerocom/io/mscw_ctm/reader.py

```diff
@@ -194,8 +196,20 @@ def __init__(self, data_id=None, data_dir=None, **kwargs):
             if isinstance(new_map, dict):
                 self.var_map.update(new_map)
             else:
-                logger.warn(f"New map {new_map} is not a dict. Skipping")
+                logger.warning(f"New map {new_map} is not a dict. Skipping")
 
+        if (pattern := kwargs.get("file_pattern", None)) is None:
+            # Pattern for the 'Base_{freq}.nc' default strategy.
+            pattern = rf"^Base_({'|'.join(self.FREQ_CODES.keys())}).nc$"
+
+        if not isinstance(pattern, str | re.Pattern):
+            raise TypeError(
+                f"Provided pattern '{pattern}' of type {type(pattern)} can't be compiled to re.Pattern. Please provide str or re.Pattern."
```

Comment:
The user only knows `file_pattern`, not `pattern`. And you haven't tried to compile yet, did you?